pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -590,6 +602,56 @@ func validateDomain(domain string, validDomains []string) bool {
 	return false
 }
 
+func validateMachinePools(newObject *hivev1.ClusterDeployment, contextLogger *log.Entry) *admissionv1beta1.AdmissionResponse {
```

Comment:
Can we do the validation here using the more ubiquitous "k8s.io/apimachinery/pkg/util/validation/field"? See the clusterprovision validator for examples.

Reply:
done